indexd/src/quic/upload.rs

```diff
@@ -346,7 +347,11 @@ impl Uploader {
         }
         read_slab_res.await??;
 
-        let object = Object::new(slabs, meta);
+        let nonce_meta = nonce_prefix
```

Comment:
Hah I definitely didn't see this coming but after reading the working in the issue it makes sense that you put it in the metadata field of the object. Sry about that but I should have chosen my words a bit more carefully. What I meant by object metadata was the `Object` type itself. So to add a `nonce` field right next to the `encryption_key` and then have the user be responsible for storing it alongside the encryption key and other metadata locally. 

Because the encryption key is already the user's responsibility and losing it will prevent us from getting the nonce so we might as well keep the nonce locally too instead of sending an encrypted version of it to the server.